build/ci/setenv

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
+
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+
+# Script to set environment variables to run CI tasks.
+#
+# Usage: setenv CONTEXT
+#
+#   CONTEXT      test or release (test by default)
```

Comment:
What is the purpose of the `test` context ? When should it be used ?

Reply:
`setenv test` should be called to run (unit or integ) tests related ci tasks and `setenv release` to run build (or e2e tests) related ci tasks. Having slept on this, it sounds like a potential source of error. I will look for a better solution.

Reply:
`setenv test` sets `GCLOUD_PROJECT` (create/delete k8s clusters).
`setenv release` sets `GO_TAGS`, `LICENSE_PUBKEY` (build the operator binary with the public license key) and `ELASTIC_DOCKER_LOGIN` (docker push ci/e2e/operator images).

We have 9 Jenkinsfile:
* build/ci/support/cleanup.jenkinsfile
* build/ci/pr/Jenkinsfile
* build/ci/e2e/custom_operator_image.jenkinsfile
* build/ci/e2e/GKE_k8s_versions.jenkinsfile
* build/ci/e2e/Jenkinsfile-aks
* build/ci/e2e/Jenkinsfile
* build/ci/e2e/stack_versions.jenkinsfile
* build/ci/e2e/vanilla_k8s.jenkinsfile
* build/ci/release/Jenkinsfile

8/9 `Jenkinsfile` use `GCLOUD_PROJECT`. The only one which doesn't need it is `e2e/Jenkinsfile-aks` for now.

7/9 build the operator with the pub key to run e2e tests or do a release.

Reply:
I removed the notion of context by decreasing the number of variable to set in 659f280.

Reply:
@pebrc Can you take a look in particular at 659f280 as it is a little sensitive? 

In a effort to simplify the Jenkinsfile, I think we can:
* move the declaration of `GO_TAGS ?= release` in the make targets `ci-e2e` and `ci-release`
* move the writing of the `export LICENSE_PUBKEY` in the `.env` file in the make target `get-elastic-public-key` that fetchs the license pub key (same for the test licenses)

Reply:
`GO_TAGS ?= release` in the make targets `ci-e2e` breaks the pr job.

The pr job is the only place where `ci-e2e` is used without the need to set `GO_TAGS` to `release` because to run the TestSmoke we don't need anything related to the licenses.

Reply:
In the same spirit of moving some logic outside the Jenkinsfile, we can add a dedicated make target `ci-e2e-smoketest` that will not have `GO_TAGS ?= release` and instead `TESTS_MATCH = TestSmoke`.

77c044f